crates/egui-wgpu/src/lib.rs

```diff
@@ -264,59 +320,60 @@ fn wgpu_config_impl_send_sync() {
 impl std::fmt::Debug for WgpuConfiguration {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         let Self {
-            supported_backends,
-            device_descriptor: _,
             present_mode,
             desired_maximum_frame_latency,
-            power_preference,
+            wgpu_device_setup: adapter_selection,
             on_surface_error: _,
         } = self;
         f.debug_struct("WgpuConfiguration")
-            .field("supported_backends", &supported_backends)
             .field("present_mode", &present_mode)
             .field(
                 "desired_maximum_frame_latency",
                 &desired_maximum_frame_latency,
             )
-            .field("power_preference", &power_preference)
+            .field("adapter_selection", &adapter_selection)
             .finish_non_exhaustive()
     }
 }
 
 impl Default for WgpuConfiguration {
     fn default() -> Self {
         Self {
-            // Add GL backend, primarily because WebGPU is not stable enough yet.
-            // (note however, that the GL backend needs to be opted-in via the wgpu feature flag "webgl")
-            supported_backends: wgpu::util::backend_bits_from_env()
-                .unwrap_or(wgpu::Backends::PRIMARY | wgpu::Backends::GL),
-
-            device_descriptor: Arc::new(|adapter| {
-                let base_limits = if adapter.get_info().backend == wgpu::Backend::Gl {
-                    wgpu::Limits::downlevel_webgl2_defaults()
-                } else {
-                    wgpu::Limits::default()
-                };
-
-                wgpu::DeviceDescriptor {
-                    label: Some("egui wgpu device"),
-                    required_features: wgpu::Features::default(),
-                    required_limits: wgpu::Limits {
-                        // When using a depth buffer, we have to be able to create a texture
-                        // large enough for the entire surface, and we want to support 4k+ displays.
-                        max_texture_dimension_2d: 8192,
-                        ..base_limits
-                    },
-                    memory_hints: wgpu::MemoryHints::default(),
-                }
-            }),
-
             present_mode: wgpu::PresentMode::AutoVsync,
 
             desired_maximum_frame_latency: None,
 
-            power_preference: wgpu::util::power_preference_from_env()
-                .unwrap_or(wgpu::PowerPreference::HighPerformance),
+            // Use "standard" wgpu adapter selection by default, which iterates adapters
+            // based on their power preference. The power preferenc can be configured from
+            // env variables.
```

Comment:
nit: it also picks a device depending on the device descriptors etc., undersells it a bit by only talking about the adapter selection